host: Panic in logmux

```
t=2015-10-31T23:37:48+0000 lvl=eror msg="error scanning log message" app=host pid=21475 host.id=host0 component=logmux fn=streamWithHistory app.id=7fa78264-6f6f-46b6-b381-efdf6bb79e23 job.id=host0-06ede156-8495-43d1-b9a7-92d1604ec378 log=/var/log/flynn/7fa78264-6f6f-46b6-b381-efdf6bb79e23.log error="MSG-LEN was longer than 5 characters"
now=2015-10-31T23:37:48+0000 fn=attach job.id=host0-06ede156-8495-43d1-b9a7-92d1604ec378 at=finish
t=2015-10-31T23:37:48+0000 lvl=info msg="request completed" component=host req_id=b1dab34a-ffda-4b52-a7b2-2754c55a07da status=101 duration=11.425161ms
panic: close of closed channel
```

https://gist.github.com/josephglanville/c17e323256a2a6b6fc9e

--- a/host/logmux/logmux.go
+++ b/host/logmux/logmux.go
@@ -546,11 +546,12 @@ func (m *Mux) streamWithHistory(appID, jobID string, follow bool, ch chan<- *rfc
 	go func() {
 		defer close(unsubscribeFn)
 		for i, name := range logs[appID] {
-			func() {
+			if err := func() (err error) {
 				l := l.New("log", name)
 				f, err := os.Open(name)
 				if err != nil {
 					l.Error("error reading log", "error", err)
+					return err
 				}
 				defer f.Close()
 				sc := bufio.NewScanner(f)
@@ -565,18 +566,17 @@ func (m *Mux) streamWithHistory(appID, jobID string, follow bool, ch chan<- *rfc
 					msg, cursor, err := utils.ParseMessage(msgCopy)
 					if err != nil {
 						l.Error("error parsing log message", "error", err)
-						continue
+						return err
 					}
 					select {
 					case msgs <- message{cursor, msg}:
 					case <-s.StopCh:
-						return
+						return nil
 					}
 				}
 				if err := sc.Err(); err != nil {
 					l.Error("error scanning log message", "error", err)
-					s.Error = err
-					close(ch)
+					return err
 				}
 				if follow && !eof && i == len(logs[appID])-1 {
 					// got EOF on last file, subscribe to stream
@@ -585,10 +585,15 @@ func (m *Mux) streamWithHistory(appID, jobID string, follow bool, ch chan<- *rfc
 					// read to end of file again
 					goto scan
 				}
-			}()
-			if !follow {
+				return nil
+			}(); err != nil {
 				close(msgs)
+				s.Error = err
+				return
 			}
+		}
+		if !follow {
+			close(msgs)
 		}
 	}()
 

